### PR TITLE
implement String.prototype.replaceAll proposal

### PR DESCRIPTION
**Pull request by targos**
Refs: https://tc39.es/proposal-string-replaceall/

### CODE REVIEW

File: src/intrinsics/StringPrototype.mjs [hunk 1]
```diff
@@ -154,29 +159,49 @@ function StringProto_includes([searchString = Value.undefined, position = Value.
...
-      if (searchStr[j] !== S[k + j]) {
-        match = false;
-        break;
+  // 8. Let searchLen be the length of searchStr.
+  const searchLen = searchStr.stringValue().length;
+  // https://tc39.es/proposal-string-replaceall/#sec-string.prototype.indexof
+  if (surroundingAgent.feature('String.prototype.replaceAll')) {
+    // Let position be ! StringIndexOf(S, searchString, start).
+    // TODO: fix bug in spect text.
+    position = X(StringIndexOf(new Value(S), searchStr, start));
```
Comment (targos) [inline src/intrinsics/StringPrototype.mjs:182]: /cc @mathiasbynens I think that's a bug in the spec.

**Comment by targos** [comment]
Added comments and fixed a spec bug. I forgot to run the whole test suite the first time.

File: src/intrinsics/StringPrototype.mjs (hunk 1 above)
Comment (mathiasbynens) [reply src/intrinsics/StringPrototype.mjs:182]: Sorry, I lack context here; what's the bug? Would you mind submitting a patch against https://github.com/tc39/proposal-string-replaceall?

Comment (mathiasbynens) [reply src/intrinsics/StringPrototype.mjs:182]: The relevant line is here: https://github.com/tc39/proposal-string-replaceall/blob/33f5e50dd27ade44a52a586933d8ec32bbee164b/spec.html#L131

Comment (targos) [reply src/intrinsics/StringPrototype.mjs:182]: I opened an issue: https://github.com/tc39/proposal-string-replaceall/issues/36

Comment (mathiasbynens) [reply src/intrinsics/StringPrototype.mjs:182]: I see. Good catch! Thank you for filing the bug :)

**Pull request merged**